### DISCUSSION TITLE
ttl of interval: 0s

### DISCUSSION THREAD

**Discussion by saifer05** [akvorado/akvorado#1036] [Q&A]
Thanks for the Akvorado!
Help me understand how this setting works:

clickhouse: => resolutions: => - interval: 0s => ttl: 192h0m0s

I expect that the raw data from the clickhouse 'flows' table will contain the last 8 days (I'm trying to adapt to the HDD resources allocated to this WM). 
But I cannot get this behavior and any HDD fills up over time. 
Tell me, what am I doing wrong?

  resolutions:
    - interval: 0s
      ttl: 192h0m0s
    - interval: 1m0s
      ttl: 168h0m0s
    - interval: 5m0s
      ttl: 2160h0m0s
    - interval: 1h0m0s
      ttl: 8760h0m0s

┌─table──────────────────────────────────────────┬─size_gb─┐
│ flows                                          │   44.12 │
│ flows_5m0s                                     │    2.09 │
│ trace_log                                      │    1.83 │
│ part_log                                       │    1.49 │
│ flows_1m0s                                     │    1.07 │
│ flows_1h0m0s                                   │    0.54 │
│ asynchronous_metric_log                        │    0.54 │
│ metric_log                                     │    0.52 │
│ query_log                                      │    0.03 │
│ .inner_id.026c5518-fe24-47a2-948f-ca9fc49d5e67 │       0 │
└────────────────────────────────────────────────┴─────────┘

VOLUME NAME                                                        LINKS     SIZE
c96efe617f93f1b25f9705ffb60682694fbd8fdd6e722193c7d39905b5d7cb01   0         0B
ca918d614d7fb4d3ad53dcb26ff8cfde82500214ebe292554e262d59c2751b8e   0         0B
akvorado_akvorado-geoip                                            2         53MB
akvorado_akvorado-run                                              1         0B
akvorado_akvorado-zookeeper                                        1         402.8MB
akvorado_akvorado-clickhouse                                       1         56.09GB
akvorado_akvorado-console-db                                       1         0B
akvorado_akvorado-kafka                                            1         14.93GB

![1](https://github.com/akvorado/akvorado/assets/16239424/2cebf95d-ce6d-4167-b193-17da6755db1c)
![2](https://github.com/akvorado/akvorado/assets/16239424/7441648d-4013-4a48-82a8-00d219e47d7d)

**Comment by vincentbernat** [comment]
Check with `show create table flows` if the TTL you provide has been applied correctly.

**Reply by saifer05** [reply]
Among the output there is this line, I believe this is it: `TTL TimeReceived + toIntervalSecond(691200)`.

**Reply by vincentbernat** [reply]
So, it should work. Note that ClickHouse only retire expired partitions. The partitions are defined by the `PARTITION BY` command, but older partitions may be bigger. By default, 15 days uses 8-hour partitions, so it should not be a problem to reclaim space. The effect is not totally immediate, but should happen quickly.

**Reply by vincentbernat** [reply]
Did you modify `max-partitions`? If too small (like 1), you'll have to wait several days.

**Reply by saifer05** [reply]
I didn’t find 'max-partitions', only this:
config/akvorado.yaml ==> kafka: => topic-configuration: => num-partitions: 4 (may have changed)
orchestrator --check --dump ==> clickhouse: => maxpartitions: 50 (has not changed)

The database was not filled for about 5 days. I had to restart, wait a little time and now the situation is as follows:

VOLUME NAME                                                        LINKS     SIZE
akvorado_akvorado-clickhouse                                       1         20.43GB
akvorado_akvorado-console-db                                       1         0B
akvorado_akvorado-kafka                                            1         1.334GB
akvorado_akvorado-geoip                                            2         45.13MB
akvorado_akvorado-run                                              1         0B
akvorado_akvorado-zookeeper                                        1         537MB
c96efe617f93f1b25f9705ffb60682694fbd8fdd6e722193c7d39905b5d7cb01   0         0B
ca918d614d7fb4d3ad53dcb26ff8cfde82500214ebe292554e262d59c2751b8e   0         0B


┌─table──────────────────────────────────────────┬─size_gb─┐
│ flows                                          │   12.08 │
│ flows_5m0s                                     │    2.09 │
│ trace_log                                      │    1.68 │
│ part_log                                       │    1.19 │
│ flows_1h0m0s                                   │    0.54 │
│ asynchronous_metric_log                        │    0.43 │
│ metric_log                                     │    0.42 │
│ flows_1m0s                                     │    0.17 │
│ query_log                                      │    0.03 │
│ .inner_id.026c5518-fe24-47a2-948f-ca9fc49d5e67 │       0 │
└────────────────────────────────────────────────┴─────────┘

I would like to request more disk space.
Increasing the "clickhouse: => maxpartitions:" parameter will change the situation with the frequency of cleaning the 'flows' table?

**Reply by vincentbernat** [reply]
The default parameter should be OK. With 8 days, 50 means that partitions are about 4 hours.

**Reply by saifer05** [reply]
Disk space has been increased for VM. According to my calculations, the service should now work for about 50-60 days. I hope Akvorado will finally become autonomous. Are there any recommendations on this issue?
